Google.PowerShell/BigQuery/BqTable.cs

```diff
@@ -330,4 +330,163 @@ public TablesResource.InsertRequest makeInsertReq()
+    /// To delete a non-empty table automatically, use the -Force parameter. If no Project is 
+    /// specified, the default project will be used. This cmdlet returns a Table object. 
+    /// This cmdlet supports the ShouldProcess function, so it has the corresponding -WhatIf 
+    /// and -Confirm flags.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> $table = Get-BqTable -Dataset "my_dataset" -Table "my_table"
+    ///   PS C:\> $table | Remove-BqTable</code>
+    ///   <para>This will remove "my_table" if it is empty, and will prompt for user confirmation 
+    ///   if it is not. All data in "my_table" will be deleted if the user accepts.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Remove-BqTable -DatasetId "my_dataset" -Table "my_table" -Force</code>
+    ///   <para>This will remove "my_table" and all of its data.</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/tables)">
+    /// [BigQuery Tables]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "BqTable", SupportsShouldProcess = true)]
+    
+    public class RemoveGcbqTable : BqCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByValue = "ByValue";
+            public const string ByDatasetObject = "ByDatasetObject";
+            public const string ByObject = "ByObject";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for tables in. If not set via PowerShell parameter processing, it will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValue)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        override public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the table that you want to get a descriptor object for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.ByValue)]
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.ByDatasetObject)]
+        public string Table { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the dataset to search. Can be a string, a Dataset, a DatasetReference, or a DatasetsData object.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.ByValue)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Dataset that you would like to search. This field takes Dataset or DatasetRefrence objects.
```

Comment:
I think you should mention something about the table that is being removed (and its connection to the dataset) in this comment.